### PR TITLE
fixes syndicate dagger

### PR DESCRIPTION
**Pull request by ZWRh3**
## About the PR 
refactors sacrifical dagger

makes its on throw stun not work in arrivals and makes its damage actually work among other fixes

adds a proc that allows overwriting the default impact sound (tried my best may need further improvements)
## Why's this needed? 
bugs bad (#9495)

### CODE REVIEW

File: code/obj/item/misc_weapons.dm [hunk 1]
```diff
@@ -486,35 +486,41 @@
...
+			var/mob/M = A
+			if (ismob(usr))
+				M:lastattacker = usr
+				M:lastattackertime = world.time
+			M.changeStatus("weakened", 6 SECONDS)
+			M.force_laydown_standup()
+			take_bleeding_damage(M, null, 5, DAMAGE_CUT)
+
+	attack(target, mob/user)
+		playsound(target,  'sound/impact_sounds/Flesh_Stab_1.ogg ', 60, 1)
```
Comment (TobleroneSwordfish) [inline code/obj/item/misc_weapons.dm:508]:
```suggestion
		playsound(target, 'sound/impact_sounds/Flesh_Stab_1.ogg', 60, 1)
```

File: code/obj/item/misc_weapons.dm [hunk 2]
```diff
@@ -486,35 +486,41 @@
...
-	..()
+
+	throw_impact(atom/A, datum/thrown_thing/thr)
+		if (..())
+			return
+		if(ismob(A))
+			var/mob/M = A
+			if (ismob(usr))
+				M:lastattacker = usr
+				M:lastattackertime = world.time
```
Comment (TobleroneSwordfish) [inline code/obj/item/misc_weapons.dm:502]:
```suggestion
				M.lastattacker = usr
				M.lastattackertime = world.time
```
both of those are defined on mob, no need to use `:`

File: code/obj/item/misc_weapons.dm [hunk 3]
```diff
@@ -486,35 +486,41 @@
...
 	force = 10.0
 	throw_range = 10
 	throwforce = 10.0
 
-/obj/item/dagger/smile/attack(mob/living/target, mob/user)
-	if(prob(10))
-		var/say = pick("Why won't you smile?","Smile!","Why aren't you smiling?","Why is nobody smiling?","Smile like you mean it!","That is not a smile!","Smile, [target.name]!","I will make you smile, [target.name].","[target.name] didn't smile!")
-		user.say(say)
-	..()
+	attack(mob/living/target, mob/user)
```
Comment (TobleroneSwordfish) [inline code/obj/item/misc_weapons.dm:519]: Absolute proc paths are generally preferred I think, better to leave it like that.

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]

File: code/obj/item/misc_weapons.dm (hunk 3 above)
Comment (Zonespace27) [reply code/obj/item/misc_weapons.dm:519]: every day a proc is made relative, an angel loses its wings

File: code/atom/throwing.dm [hunk 4]
```diff
@@ -37,28 +37,34 @@
...
 	if(SEND_SIGNAL(src, COMSIG_MOVABLE_HIT_THROWN, hit_atom, thr))
 		return
 	if(SEND_SIGNAL(hit_atom, COMSIG_ATOM_HITBY_THROWN, src, thr))
 		return
 	var/impact_sfx = hit_atom.hitby(src, thr)
+	var/overwritten_impact_sfx = src.overwrite_impact_sfx(impact_sfx,hit_atom, thr)
+	impact_sfx = overwritten_impact_sfx
 	if(src && impact_sfx)
 		playsound(src, impact_sfx, 40, 1)
+	return FALSE
```
Comment (TobleroneSwordfish) [inline code/atom/throwing.dm:67]:
```suggestion
```
not needed, and we return null from the signals above so may as well here

**Review by TobleroneSwordfish — APPROVED** [review]
remind me to buy some of these next rev round, didn't know they were a 6 second stun

File: code/obj/item/misc_weapons.dm (hunk 3 above)
Comment (TobleroneSwordfish) [reply code/obj/item/misc_weapons.dm]: *hurriedly hides several hundred lines of relative proc commits*

File: code/atom/throwing.dm [hunk 5]
```diff
@@ -37,26 +37,31 @@
 		src.pixel_x = text2num(params["icon-x"]) - 16
 		src.pixel_y = text2num(params["icon-y"]) - 16
 
+/atom/movable/proc/overwrite_impact_sfx(original_sound,hit_atom,thr)
```
Comment (ZeWaka) [inline code/atom/throwing.dm:40]:
```suggestion
/atom/movable/proc/overwrite_impact_sfx(original_sound, hit_atom, thr)
```

File: code/atom/throwing.dm [hunk 6]
```diff
@@ -37,26 +37,31 @@
...
 
-	reagents?.physical_shock(20)
+	src.reagents?.physical_shock(20)
 	if(SEND_SIGNAL(src, COMSIG_MOVABLE_HIT_THROWN, hit_atom, thr))
 		return
 	if(SEND_SIGNAL(hit_atom, COMSIG_ATOM_HITBY_THROWN, src, thr))
 		return
 	var/impact_sfx = hit_atom.hitby(src, thr)
+	var/overwritten_impact_sfx = src.overwrite_impact_sfx(impact_sfx,hit_atom, thr)
+	impact_sfx = overwritten_impact_sfx
```
Comment (ZeWaka) [inline code/atom/throwing.dm:64]: no just combine these why such a useless var

File: code/obj/item/misc_weapons.dm [hunk 7]
```diff
@@ -486,22 +486,28 @@
 		..()
 		BLOCK_SETUP(BLOCK_KNIFE)
 
+/obj/item/dagger/overwrite_impact_sfx(original_sound, hit_atom, thr)
+	..()
```
Comment (ZeWaka) [inline code/obj/item/misc_weapons.dm:490]:
```suggestion
	. = ..()
```
:)

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/atom/throwing.dm (hunk 6 above)
Comment (ZWRh3) [reply code/atom/throwing.dm:64]: because of the old behavior which actually did use 2 vars whoops

Comment (ZeWaka) [reply code/atom/throwing.dm:64]: so, combine them.

Comment (ZWRh3) [reply code/atom/throwing.dm:64]: i will i was just going to sleep right as i read that

**Pull request merged**